src/lib.rs

```diff
@@ -20,18 +20,17 @@ use std::thread;
 pub use color::Color;
 pub use standard_out_api::parse_command_string;
 pub use standard_out_api::run_commands_stdout;
-pub use standard_out_api::run_commands_stdout_with_options;
 pub use standard_out_api::ConsoleCommand;
 
 pub use writer_api::run_commands_writer;
-pub use writer_api::run_commands_writer_with_options;
 
 #[derive(Debug)]
 pub enum CommandError {
     EmptyCommand,
     ParseError(String),
 }
 
+#[derive(Clone)]
```

Comment:
Just out of curiosity, why do you implement `Clone` here? Doesn't this hide where code using this is making clones?

I'm not questioning your decision here. I'm trying to undersntand when to implement `Clone` and when not, and this seems like an example when it makes sense.

Reply:
Adding clone allows the caller to pass references into the runner rather than owned structures.
This command is then cloned internally.

Allowing callers to pass in references allows them to quickly duplicate commands with small modifications easily. 

`run_commands_stdout(runner::new().command(cmd).command(cmd.color(BLUE)).command(cmd.env("hello", "world"))`

Reply:
Couldn't this be implemented by having the `command` method consume `self` and return `self`?

Reply:
these methods do return a mutable reference to self. 
In the same pattern as `std::process::Command` which I think is the right solution here.

As for Clone, the two methods: `run_commands` and `run_commands_stdout` take an `&Runner` which means that we need to clone each underlying `Command` in order to `Send` it to the underlying runner threads that run each command.

Having the API take an `&runner` allows flexibility in the interface to pass either pointers or structures. However, if we want to just accept an owned `Runner` we can get rid of `Clone` implementation on these types. what do you think?